### ISSUE TITLE
 Vagrant was unable to mount VirtualBox shared folders - OpenSUSE

### ISSUE THREAD

**Issue by Gsantomaggio** [chef/bento#1185]
# Version:

```
➜ vboxmanage --version
6.0.4r128413
➜  leap_bento vagrant version
Installed Version: 2.2.4
Latest Version: 2.2.4


chef --version
Chef Development Kit Version: 3.8.14
chef-client version: 14.10.9
delivery version: master (9d07501a3b347cc687c902319d23dc32dd5fa621)
berks version: 7.0.7
kitchen version: 1.24.0
inspec version: 3.6.6

```

# Environment: 
```
lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.2 LTS
Release:        18.04
Codename:       bionic
```
# Scenario:

Tryin to use it inside a receipe

# Steps to Reproduce:
Tryin to use it inside a receipe:

```
driver:
  name: <%= ENV['KI_DRIVER'] || 'vagrant' %>

verifier:
  name: inspec

provisioner:
  name: chef_zero
  product_name: chef
  install_strategy: always

transport:
  name: "ssh"

platforms:
  - name: opensuse-leap-15.0

 - name: rabbitmq-erlang-latest-rpm-suse
        includes: ["opensuse-leap-15.0"]
```

# Expected Result:

The opensuse leap 15 running machine

# Actual Result:
```
STDERR: Vagrant was unable to mount VirtualBox shared folders. This is usually
because the filesystem "vboxsf" is not available. This filesystem is
made available via the VirtualBox Guest Additions and kernel module.
Please verify that these guest additions are properly installed in the
guest. This is not a bug in Vagrant and is usually caused by a faulty
Vagrant box. For context, the command attempted was:

mount -t vboxsf -o uid=1000,gid=1000 tmp_omnibus_cache /tmp/omnibus/cache
```

I tried other platforms, and it works without problems.
Thank you

**Comment by dcermak** [comment]
I have seen similar errors when building vagrant boxes, when I forgot to load the `vboxfs` kernel module. In my boxes I usually add this line:
``` shell
echo vboxsf > /etc/modules-load.d/vboxsf.conf
```
to a provisioner script.

I wanted to give this a try for the Leap 15 box, but was unfortunately unable to build it on Tumbleweed with packer.

**Comment by Gsantomaggio** [comment]
I rebuilt the image locally and it works. 

With packer:
```
packer build -only=virtualbox-iso opensuse-leap-15.0-x86_64.json
```
then:
```
vagrant box add --name OpenSUSE_Leap15_local /home/gabriele/git/Chef/bento/builds/opensuse-leap-15.0.virtualbox.box
vagrant up && vagrant ssh
....
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Machine already provisioned. Run `vagrant provision` or use the `--provision`
==> default: flag to force provisioning. Provisioners marked to run always will still run.
Last failed login: Mon Mar 25 15:17:03 UTC 2019 from 10.0.2.2 on ssh:notty
There were 8 failed login attempts since the last successful login.
Last login: Mon Mar 25 14:42:06 2019 from 10.0.2.2
Have a lot of fun...
```

and it works, maybe it is enough to rebuild the image. 

Is there a way to do that?

**Comment by mgruner** [comment]
I am seeing the same behaviour with the current opensuse-leap-15.1 box (v201910.20.0).

With "vagrant up", I get:

```
==> default: Mounting shared folders...
    default: /vagrant => /private/tmp/suse
Vagrant was unable to mount VirtualBox shared folders. This is usually
because the filesystem "vboxsf" is not available. This filesystem is
made available via the VirtualBox Guest Additions and kernel module.
Please verify that these guest additions are properly installed in the
guest. This is not a bug in Vagrant and is usually caused by a faulty
Vagrant box. For context, the command attempted was:

mount -t vboxsf -o uid=1000,gid=1000 vagrant /vagrant

The error output from the command was:



```

Logging into the machine shows that vboxsf is already loaded:

```
opensuse-leap-x64:~ # lsmod | grep vbox
vboxsf                 86016  0
vboxvideo              49152  1
drm_kms_helper        204800  1 vboxvideo
ttm                   126976  1 vboxvideo
vboxguest             344064  1 vboxsf
drm                   491520  4 vboxvideo,ttm,drm_kms_helper
```

However, the mounting still fails from within the machine too:

```
opensuse-leap-x64:~ # mount -t vboxsf -o uid=1000,gid=1000 vagrant /vagrant
mount: /vagrant: wrong fs type, bad option, bad superblock on vagrant, missing codepage or helper program, or other error.
```

I see the exact same behaviour when rebuilding the machine locally. This renders the box unusable at present.

**Comment by mgruner** [comment]
`dmesg` tells me that the version loaded is from SUSE: 

```
[   16.385429] vboxsf: Successfully loaded version 6.0.12_SUSE
[   16.385462] vboxsf: Successfully loaded version 6.0.12_SUSE on 4.12.14-lp151.28.13-default SMP mod_unload modversions  (LINUX_VERSION_CODE=0x40c0e)
[   16.385660] vboxsf: Invalid info signature: 0x75 0x69 0x64 0x3d!
[   16.385661] vbsf_read_super_aux err=-22
[   16.475851] vboxsf: Invalid info signature: 0x75 0x69 0x64 0x3d!
[   16.475854] vbsf_read_super_aux err=-22
[   21.571625] vboxsf: Invalid info signature: 0x75 0x69 0x64 0x3d!
[   21.571627] vbsf_read_super_aux err=-22
[   26.666942] vboxsf: Invalid info signature: 0x75 0x69 0x64 0x3d!
[   26.666944] vbsf_read_super_aux err=-22
[  191.490976] vboxsf: Invalid info signature: 0x75 0x69 0x64 0x3d!
[  191.490979] vbsf_read_super_aux err=-22
```

Could it be that there is a conflict between the version from VirtualBox that is installed by packer and another one that comes from SUSE?

```
opensuse-leap-x64:/lib # find . -name "vboxsf*"
./modules/4.12.14-lp151.27-default/weak-updates/extra/vboxsf.ko
./modules/4.12.14-lp151.28.20-default/weak-updates/extra/vboxsf.ko
./modules/4.12.14-lp151.28.13-default/extra/vboxsf.ko
./modules/4.12.14-lp151.28.10-default/extra/vboxsf.ko

opensuse-leap-x64:/lib # uname -a
Linux opensuse-leap-x64 4.12.14-lp151.28.20-default #1 SMP Tue Oct 8 05:42:34 UTC 2019 (2982b5d) x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by woky** [comment]
I've encountered error messages in dmesg that https://github.com/chef/bento/issues/1185#issuecomment-553381519 reports. My box is Debain testing and I've installed guest additions from Debian repositories:

```
apt install -y virtualbox-guest-dkms
```

The solution for me was to:
```
apt install -y virtualbox-guest-utils
systemctl enable --now virtualbox-guest-utils
```

While `vboxsf` module were loaded already, the above service also starts up `VBoxService`. That made mount work in my case.
